go/tools/builders/compile.go

```diff
@@ -19,30 +19,45 @@ package main
 import (
 	"fmt"
 	"go/build"
+	"go/parser"
+	"go/token"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
+	"strings"
 )
 
 func run(args []string) error {
 	// process the args
 	if len(args) < 2 {
-		return fmt.Errorf("Usage: compile gotool [sources] -- <extra options>")
+		return fmt.Errorf("Usage: compile gotool [-dep importpath ...] [sources] -- <extra options>")
 	}
 	gotool := args[0]
 	args = args[1:]
 	sources := []string{}
 	goopts := []string{}
+	deps := []string{}
 	bctx := build.Default
 	bctx.CgoEnabled = true
+	depNext := false
+argsLoop:
 	for i, s := range args {
```

Comment:
This has become too much like custom flag parsing for my liking
How about just use the flags package, the -- will force it to stop processing args anyway, so we can consume flags.Args() into the goops easily.

You will want something like the below to allow you to consume mutiple -dep args into a single list, and maybe do the same for srcs, probably put it in a separate flags.go so we can can use the same approach in all the builders.

```
// multiFlag allows repeated string flags to be collected into a slice
type multiFlag struct {
	values *[]string
}

func (m multiFlag) String() string {
	if len(*m.values) == 0 {
		return ""
	}
	return fmt.Sprint(*m.values)
}

func (m multiFlag) Set(v string) error {
	(*m.values) = append(*m.values, v)
	return nil
}
```